packages/roosterjs-editor-api/lib/utils/execCommand.ts

```diff
@@ -25,11 +26,10 @@ export default function execCommand(editor: IEditor, command: DocumentCommand) {
     if (range && range.collapsed) {
         editor.addUndoSnapshot();
         formatter();
-
         if (isPendableFormatCommand(command)) {
             // Trigger PendingFormatStateChanged event since we changed pending format state
             editor.triggerPluginEvent(PluginEventType.PendingFormatStateChanged, {
-                formatState: getPendableFormatState(editor.getDocument()),
+                formatState: setPendableFormatState(editor.getPendingFormatState(), command),

```

Comment:
Here you need to get the format state before running formatter(), then set it to opposite value. Otherwise we rely on an assumption that formatter() will not change DOM structure. Although it is true, but we should not rely on it.

```
const formatState = editor.getPendingFormatState();
formatter();

const formatName = Object.keys(PendableFormatCommandMap).filter(
        (x: PendableFormatNames) => PendableFormatCommandMap[x] == command
    )[0];

if (formatName ) {
   fomatState[formatName ] = !formatState[formatName ];
    editor.triggerPluginEvent(...)
}
```